### PR TITLE
Issue/264 bug null formula

### LINKED ISSUE #264: [bug] bmf(y ~ 1) + formula(NULL) gives error
**Describe the bug**

``` r
library(bmm)
bmf(y ~ 1) + formula(NULL)
#> Error in object[[length(object)]]: attempt to select less than one element in integerOneIndex
```

<sup>Created on 2025-02-17 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

The same is fine with brms formula:

``` r
library(brms)
#> Loading 'brms' package (version 2.22.0).
bf(y ~ 1) + formula(NULL)
#> y ~ 1
```

<sup>Created on 2025-02-17 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Expected behavior**
Should return the initial formula, as in brms, instead of giving an error

### PR DESCRIPTION
**Pull request by venpopov**
#### Summary

closes #264 and updates NEWS. Also add a meaningful error message that bmf( ~ something) is not allowed - formulas must have a left-hand-side variable. Previously we had a cryptic error 

#### Tests

[x] Confirm that all tests passed
[x] Confirm that devtools::check() produces no errors

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # bmm (development version)
 
-# bmm 1.1.0
+# bmm 1.1.1
```
Comment (GidonFrischkorn) [inline NEWS.md:3]: Given that we have not yet released version 1.1.0 does it make sense to set this to 1.1.1. given that this is only on develop?

File: NEWS.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 
 ### New models
-* Add the Memory Measurement Model (Oberauer & Lewandowsky, 2019) as new model class **m3** with three versions: simple span (**ss**), complex span (**cs**), and **custom**. For details, see the [article](https://venpopov.github.io/bmm/articles/bmm_m3.html) on the `bmm` website (#237). Thanks to @GidonFrischkorn and @chenyu-psy
+* Add the Memory Measurement Model (Oberauer & Lewandowsky, 2019) and its generalization as the Multinomial Measurement Model for categorical decision tasks as new model class **m3** with three versions: simple span (**ss**), complex span (**cs**), and **custom**. For details, see the [article](https://venpopov.github.io/bmm/articles/bmm_m3.html) on the `bmm` website (#237). Thanks to @GidonFrischkorn and @chenyu-psy
```
Comment (GidonFrischkorn) [inline NEWS.md:6]: Given that I am an author of the package I think we can only than Chenyu for his contributions towards implementing the model. Or do you think we should also highlight our contributions for implementing new models, so that it is more visible who contributed towards the implementation of which models?

**Review by GidonFrischkorn — APPROVED** [review]
This looks good. I tested the new functionality myself and saw that you also added some tests to check if adding a `NULL` formula works properly. 

I only had two minor comments regarding the version number and contribution statements in the `NEWS.md`

**Comment by venpopov** [comment]
Whatever you prefer! I think the first makes sense - to me it’s not important to highlight when one of us has contributed more or less, especially since we always review each others work and help with revisions. I wrote it as it is now without thinking about it

**Comment by venpopov** [comment]
> Whatever you prefer! I think the first makes sense - to me it’s not important to highlight when one of us has contributed more or less, especially since we always review each others work and help with revisions. I wrote it as it is now without thinking about it
> […](#)
> On Feb 18, 2025 at 9:56 AM +0100, Gidon Frischkorn ***@***.***>, wrote: @GidonFrischkorn commented on this pull request. In NEWS.md: > ### New models -* Add the Memory Measurement Model (Oberauer & Lewandowsky, 2019) as new model class **m3** with three versions: simple span (**ss**), complex span (**cs**), and **custom**. For details, see the [article](https://venpopov.github.io/bmm/articles/bmm_m3.html) on the `bmm` website (#237). Thanks to @GidonFrischkorn and @chenyu-psy +* Add the Memory Measurement Model (Oberauer & Lewandowsky, 2019) and its generalization as the Multinomial Measurement Model for categorical decision tasks as new model class **m3** with three versions: simple span (**ss**), complex span (**cs**), and **custom**. For details, see the [article](https://venpopov.github.io/bmm/articles/bmm_m3.html) on the `bmm` website (#237). Thanks to @GidonFrischkorn and @chenyu-psy Given that I am an author of the package I think we can only than Chenyu for his contributions towards implementing the model. Or do you think we should also highlight our contributions for implementing new models, so that it is more visible who contributed towards the implementation of which models? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***>

Feel free to change the news in your preferred way and merge into develop
